datafusion/expr-common/src/signature.rs

```diff
@@ -146,6 +146,8 @@ pub enum ArrayFunctionSignature {
     /// Specialized Signature for MapArray
     /// The function takes a single argument that must be a MapArray
     MapArray,
+    /// Specialized Signature for MapExtract and similar functions
+    MapArrayAndKey,
```

Comment:
Would it be possible to use `TypeSignature::UserDefined` to avoid enumeration bloat and make it more cohesive?

Reply:
Only if there are more than 1 functions has the same signature, we could add it